ci/checks/black_lists.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+# Copyright (c) 2019, NVIDIA CORPORATION.
+##########################################
+# cuML black listed function call Tester #
+##########################################
+
+TARGET_BRANCH="master"
+
+# Checkout master for comparison
+git checkout --quiet $TARGET_BRANCH
+
+# Switch back to tip of PR branch
+git checkout --quiet current-pr-branch
+
+# Ignore errors during searching
+set +e
+
+# Disable history expansion to enable use of ! in perl regex
+set +H
+
+RETVAL=0
+
+for black_listed in cudaDeviceSynchronize cudaMalloc cudaMallocManaged cudaFree cudaMallocHost cudaHostAlloc cudaFreeHost; do
+    TMP=`git --no-pager diff --ignore-submodules -w --minimal -U0 -S"$black_listed" $TARGET_BRANCH | grep '^+' | grep -v '^+++' | grep "$black_listed"`
```

Comment:
@jirikraus I just realized a potential case of false-alarm while running this script for a set of local changes I've been making. I had a line with a comment mentioning about `cudaMalloc` and this script yelled at me for this case :(.

Reply:
That's a good point. Given the various ways comments could be formed seems like detecting those cases reliably requires a more sophisticated infrastructure than a a shell script with git and grep. For now I added a check that C++ style comments with only white spaces preceding the comment are filtered out. Do you think that is sufficient till we have a more sophisticated infrastructure in place?

Reply:
Yes... it sounds like we need a more sophisticated infra for catching such use-cases. For now, I think this approach is good enough.